go/keybase/main.go

```diff
@@ -37,6 +38,34 @@ type Stopper interface {
 }
 
 func main() {
+	memprofile := os.Getenv("KB_MEM_PROFILE")
+	if len(memprofile) > 0 {
+		runtime.MemProfileRate = 1
+	}
+
+	defer func() {
+		if len(memprofile) > 0 {
+			f, err := os.Create(memprofile)
+			if err != nil {
+				fmt.Println(err)
+				os.Exit(1)
+			}
+			pprof.WriteHeapProfile(f)
+			f.Close()
+		}
+	}()
+
+	cpuprofile := os.Getenv("KB_CPU_PROFILE")
```

Comment:
is there any way to pull this out of the main function, and just pass back a function to be defer'ed until the end?  if we can keep the main function as tight as possible it will help with code cleanliness.